### PR TITLE
SA-1982: Add segment tracking on creating scheduled report

### PR DESCRIPTION
**Pull request by kchu1337**
### What Changed

- Added Segment tracking for Creating a scheduled report

### How To Test

- Create a scheduled report; `/signals/schedule/bounce`
- Go to https://app.segment.com/sparkpost-daniel-chalef/sources/webui_dev/debugger
- See if the event showed up.

### To Do

- [ ] MixPanel Check?
- [x] Test?

### CODE REVIEW

File: src/pages/reportBuilder/ScheduledReportCreatePage.js [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 import { Page } from 'src/components/matchbox';
 import {
   ScheduledReportDetailsForm,
   ScheduledReportTimingForm,
 } from './components/ScheduledReportForm';
 import { Loading } from 'src/components/loading';
+import { PageLink } from 'src/components/links';
+import RedirectAndAlert from 'src/components/globalAlert/RedirectAndAlert';
+import { getReport, createScheduledReport } from 'src/actions/reports';
+import { showAlert } from 'src/actions/globalAlert';
```
Comment (kchu1337) [inline src/pages/reportBuilder/ScheduledReportCreatePage.js:15]: Just moved some stuff around

File: cypress/tests/integration/signals-analytics/analytics-report/savedReports.spec.js [hunk 2]
```diff
@@ -176,7 +176,6 @@ if (IS_HIBANA_ENABLED) {
     it('removes report id if saving a new report using an existing report', () => {
       cy.visit(`${PAGE_URL}&report=d50d8475-d4e8-4df0-950f-b142f77df0bf`);
       cy.wait([
-        '@accountReq',
```
Comment (kchu1337) [inline cypress/tests/integration/signals-analytics/analytics-report/savedReports.spec.js:179]: Just so tests pass.

**Review by nicklemmon — COMMENTED** [review]
@kchu1337 You probably weren't ready for another pass, but I left a few comments here for you that might help on the testing side of things.

The request is happening and the data is being sent to Segment it looks like.

Is there any way to verify things are handled by Mixpanel as expected? It looks like maybe not in staging.

File: cypress/support/commands.js [hunk 3]
```diff
@@ -131,6 +131,12 @@ Cypress.Commands.add('stubAuth', ({ hasRefreshToken = false } = {}) => {
     status: 200,
     response: {},
   }).as('stubbedSockNodeGet');
+  cy.route({
+    method: 'POST',
+    url: 'v1/*',
```
[the remark this answers is not in the document]
Comment (kchu1337) [reply cypress/support/commands.js:136]: Not sure if I can. In the route just shows up as `v1/*`

[the remark this answers is not in the document]
Comment (nicklemmon) [reply cypress/support/commands.js:136]: It worked for me locally? 🤷

[the remark this answers is not in the document]
Comment (kchu1337) [reply cypress/support/commands.js]: Updated cypress and it worked. Maybe is was in the intercept update and I just didnt update my local node modules

**Review by nicklemmon — APPROVED** [review]
@kchu1337 👍 Looks good to me. Nice work on all of the additional testing.

File: src/pages/reportBuilder/helpers/tests/scheduledReports.test.js [hunk 4]
```diff
@@ -1,141 +1,194 @@
...
-  };
-
-  it('formats form values for daily schedule correctly', () => {
-    const formValues = {
-      ...formValuesBase,
-      day: undefined,
-      period: 'AM',
-      time: '12:00',
-      timing: 'daily',
+describe('ScheduledReports helpers', () => {
```
Comment (kchu1337) [inline src/pages/reportBuilder/helpers/tests/scheduledReports.test.js:8]: Added some more layers. The other changes are mostly whitespace stuff.

**Pull request merged**